homeassistant/components/sensor/__init__.py

```diff
@@ -713,6 +714,18 @@ def state(self) -> Any:
                 report_issue,
             )
 
+        # Validate unit of measurement used for sensors with a state class
+        if (
+            state_class
+            and (units := STATE_CLASS_UNITS.get(state_class)) is not None
+            and native_unit_of_measurement not in units
+        ):
+            raise ValueError(
+                f"Sensor {self.entity_id} ({type(self)}) is using native unit of "
+                f"measurement '{native_unit_of_measurement}' which is not a valid unit "
+                f"for the state class ('{state_class}') it is using; expected one of {units};"
+            )
```

Comment:
This seems like it could be done in another PR?

Reply:
I had the same comment. @edenhaus should we just move it to another PR then, I don't think it's a lot of extra work?

Reply:
https://github.com/home-assistant/core/pull/141444